source/credentials_utils.c

```diff
@@ -275,6 +275,7 @@ struct aws_credentials *aws_parse_credentials_from_json_document(
     struct aws_json_value *document_root = aws_json_value_new_from_string(allocator, document);
     if (document_root == NULL) {
         AWS_LOGF_ERROR(AWS_LS_AUTH_CREDENTIALS_PROVIDER, "Failed to parse document as Json document.");
+        aws_raise_error(AWS_ERROR_INVALID_ARGUMENT);
```

Comment:
we should fix this is json.c
aws_json_value_new_from_string() ought to raise an error
and that error should be AWS_ERROR_INVALID_JSON, which is more useful than INVALID_ARGUMENT